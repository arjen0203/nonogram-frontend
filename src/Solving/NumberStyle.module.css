.ToBeSolvedTop {
    border-left: 2px solid #35393C;
    border-right: 2px solid #35393C;
    background-color: rgba(0, 0, 0, 0.08);
    color: #35393C;
    height: 34px;
    margin: -1px;
    padding: 0;
    width: 34px;
    text-align: center;
    font-size: 20px;
    line-height: 32px;
}

.SolvedTop {
    border-left: 2px solid #5C6469;
    border-right: 2px solid #5C6469;
    background-color: rgba(0, 0, 0, 0.08);
    color: rgba(53, 57, 60, 0.25);
    height: 34px;
    margin: -1px;
    padding: 0;
    width: 34px;
    text-align: center;
    font-size: 20px;
    line-height: 32px;
}

.ToBeSolvedSide {
    border-top: 2px solid #35393C;
    border-bottom: 2px solid #35393C;
    background-color: rgba(0, 0, 0, 0.08);
    color: #35393C;
    height: 34px;
    margin: -1px;
    padding: 0;
    width: 34px;
    text-align: center;
    font-size: 20px;
    line-height: 32px;
}

.SolvedSide {
    border-top: 2px solid #5C6469;
    border-bottom: 2px solid #5C6469;
    background-color: rgba(0, 0, 0, 0.08);
    color: rgba(53, 57, 60, 0.25);
    height: 34px;
    margin: -1px;
    padding: 0;
    width: 34px;
    text-align: center;
    font-size: 20px;
    line-height: 32px;
}