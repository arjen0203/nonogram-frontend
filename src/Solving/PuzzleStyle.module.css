.row {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    padding-right: 50vw;
}

.topNumberColumn{
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
}