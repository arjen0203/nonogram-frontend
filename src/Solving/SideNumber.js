import React from "react";
import SolveNumberStyle from './NumberStyle.module.css';


class SideNumber extends React.Component {
    render() {
        return (
            <div className={SolveNumberStyle.ToBeSolvedSide}>{this.props.value}</div>
        );
    }
}

export default SideNumber;