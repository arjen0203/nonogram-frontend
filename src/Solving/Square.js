import React from "react";

class Square extends React.Component {
    render() {
        var display;
        var styling = "square";
        if (this.props.value === 1){
            styling = "square-filled";
        } else if (this.props.value === 2){
            display = "âœ–";
        }
        return (
            <div data-testid="square"
                className={styling}
                onMouseDown={(e) => this.props.onMouseDown(e)}
                onContextMenu={(e) => this.props.onContextMenu(e)}
                onMouseEnter={(e) => this.props.onMouseEnter(e)}
            >
                {display}
            </div>
        );
    }
}

export default Square;