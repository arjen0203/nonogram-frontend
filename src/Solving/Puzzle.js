import React from "react";
import Square from "./Square";
import PuzzleStyle from "./PuzzleStyle.module.css";
import TopNumber from "./TopNumber";
import SideNumber from "./SideNumber";

class Puzzle extends React.Component {
    createBoxes() {

        const x = this.props.topRow.length;
        const y = this.props.sideRow.length;

        var items = [];

        var numbers = [];
        //top number hints
        for (var m = 0; m < x; m++){
            var sameColumn = [];

            for (var n = 0; n < this.props.topRow[m].length; n++){
                sameColumn.push(<TopNumber key={["T",n,m]} value={this.props.topRow[m][n]}></TopNumber>)
            }

            if (sameColumn.length === 0){
                sameColumn.push(<TopNumber key={["T",n,m]}></TopNumber>)
            }

            numbers.push(<div key={["TN",m]} className={PuzzleStyle.topNumberColumn}>{sameColumn}</div>);
        }
        items.push(<div key={"TopNumbers"} className={PuzzleStyle.row}> {numbers} </div>);

        //cubes and side number hints
        for (var i = 0; i < y; i++){
            var row = [];

            for (var q = 0; q < this.props.sideRow[i].length; q++){
                row.push(<SideNumber key={["S",i,q]} value={this.props.sideRow[i][q]}></SideNumber>)
            }
            if (row.length === 0){
                row.push(<SideNumber key={["S",i,q]}></SideNumber>)
            }

            for (var j = 0; j < x; j++){
                const x = i;
                const y = j;
                row.push(<Square key={["C",i,j]}
                    value={this.props.solveState[i][j]}
                                 onMouseDown={(e) => this.onClick(x, y, e)}
                                 onContextMenu={(e) => this.noContextMenu(e)}
                                 onMouseEnter={(e) => this.onDrag(x, y, e)}
                ></Square>);
            }
            items.push(<div key={["CR",i]} className={PuzzleStyle.row}> {row} </div>);
        }

        return (<div>{items}</div>);
    }

    noContextMenu(e){
        e.preventDefault();
    }

    onClick(x,y,e) {
        e.preventDefault();

        const solveState = this.props.solveState.slice();

        var importantChange = false;

        if (e.nativeEvent.which === 1) {
            if (solveState[x][y] === 1) solveState[x][y] = 0;
            else solveState[x][y] = 1;
            importantChange = true;
        } else if (e.nativeEvent.which === 3) {
            if (solveState[x][y] === 2) {
                solveState[x][y] = 0;
            }
            else if (solveState[x][y] === 1) {
                solveState[x][y] = 2;
                importantChange = true;
            } else {
                solveState[x][y] = 2;
            }
        }

        this.setState({solveState: solveState});

        if (importantChange) this.checkNumbers(x, y);
    }

    onDrag(x,y,e) {
        e.preventDefault();

        var importantChange = false;

        const solveState = this.props.solveState.slice();

        if (e.nativeEvent.which === 1 && solveState[x][y] === 0) {
            solveState[x][y] = 1;
            importantChange = true;
        } else if (e.nativeEvent.which === 3 && solveState[x][y] === 0) {
            solveState[x][y] = 2;
        } else {
            return;
        }

        this.setState({solveState: solveState});

        if (importantChange) this.checkNumbers(x, y);
    }

    checkNumbers(y,x) {
        var numbersTop = this.props.topRow[x];
        var currentFIllColumn = [];

        var counter = 0;
        var allTrue = true;

        //gets the values of the column
        for (var i = 0; i < this.props.solveState.length; i++) {
            if (this.props.solveState[i][x] === 1) {
                counter++;
            } else if (counter !== 0) {
                currentFIllColumn.push(counter);
                counter = 0;
            }
        }

        if (counter !== 0 || currentFIllColumn.length === 0) {
            currentFIllColumn.push(counter);
            counter = 0;
        }

        for (var k = 0; k < numbersTop.length; k++) {
            if (currentFIllColumn[k] === numbersTop[k].number && currentFIllColumn.length <= numbersTop.length) {
                numbersTop[k].holds = true;
            } else {
                numbersTop[k].holds = false;
                allTrue = false;
            }
        }

        var numbersSide = this.props.sideRow[y];
        var currentFIllRow = [];

        //gets the values of the row
        for (var j = 0; j < this.props.solveState[0].length; j++) {
            if (this.props.solveState[y][j] === 1) {
                counter++;
            } else if (counter !== 0) {
                currentFIllRow.push(counter);
                counter = 0;
            }
        }
        if (counter !== 0 || currentFIllRow.length === 0) currentFIllRow.push(counter);


        for (var l = 0; l < numbersSide.length; l++) {
            if (currentFIllRow[l] === numbersSide[l].number && currentFIllRow.length <= numbersSide.length) {
                numbersSide[l].holds = true;
            } else {
                numbersSide[l].holds = false;
                allTrue = false;
            }
        }
        if (allTrue) this.checkSolved();
    }

    checkSolved() {
        for(var x = 0; x < this.props.topRow.length; x++){
            for (var i = 0; i < this.props.topRow[x].length; i++){
                if (this.props.topRow[x][i].holds === false) {
                    return;
                }
            }
        }
        for(var y = 0; y < this.props.sideRow.length; y++){
            for (var j = 0; j < this.props.sideRow[y].length; j++){
                if (this.props.sideRow[y][j].holds === false) return;
            }
        }
        alert("YOU SOLVED IT!!!!");
    }

    render() {
        return (
            <div>
                {this.createBoxes()}
            </div>
        );
    }
}

export default Puzzle;