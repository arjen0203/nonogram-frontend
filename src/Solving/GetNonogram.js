import React from "react";
import Puzzle from "./Puzzle";

class GetNonogram extends React.Component {
    state = {
        loading: true,
        solveState: null,
        topRow: null,
        sideRow: null,
        error: ""
    };

    componentWillUnmount() {
        this._isMounted = false;
    }

    async componentDidMount() {
        this._isMounted = true;

        const URL = 'https://nonograms.nl/api/nonogram/get?id=' + this.props.id;

        fetch(URL)
            .then( async res => {
                if (res.ok) {
                    res.json().then(data => {
                        this.setRows(data.topValues, data.sideValues)
                        this.setState({loading: false})
                    })
                } else {
                    const error = await res.text();
                    this.setState({error});
                }
            }).catch(() => this.setState({error: "Could not communicate with server"}));

    }

    setRows(topRow, sideRow){
        var newTopRow = [];

        for (let i = 0; i < topRow.length; i++){
            newTopRow.push([])
            for (let j = 0; j < topRow[i].length; j++){
                let isZero = false;
                if (topRow[i][j] === 0) isZero = true;
                newTopRow[i].push({number: topRow[i][j], holds: isZero});
            }
        }

        var newSideRow = [];

        for (let i = 0; i < sideRow.length; i++){
            newSideRow.push([])
            for (let j = 0; j < sideRow[i].length; j++){
                let isZero = false;
                if (sideRow[i][j] === 0) isZero = true;
                newSideRow[i].push({number: sideRow[i][j], holds: isZero});
            }
        }

        var newSolveState = [];

        for (let i = 0; i < sideRow.length; i++){
            newSolveState.push([])
            for (let j = 0; j < topRow.length; j++){
                newSolveState[i].push(0);
            }
        }
        if (this._isMounted) {
            this.setState({topRow: newTopRow, sideRow: newSideRow, solveState: newSolveState});
        }
    }

    render() {
        return (
            <div>
                {this.state.loading ? (
                    <div>{this.state.error}</div>
                    ) : (
                        <Puzzle topRow={this.state.topRow} sideRow={this.state.sideRow} solveState={this.state.solveState}> </Puzzle>
                )}
            </div>
        );
    }
}

export default GetNonogram;