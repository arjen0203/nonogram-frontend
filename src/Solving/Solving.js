import React from "react";
import Puzzle from "./Puzzle";

class Solving extends React.Component {
    render() {
        const top = [[[1, false]], [[5, false]], [[2, false]], [[5, false]], [[2, false],[1, false]], [[2, false]]];
        const side = [[[2, false],[1, false]], [[1,false],[3,false]], [[1,false],[2,false]], [[3,false]], [[4,false]], [[1,false]]];

        const progress = [[0,0,0,0,0,0],
                          [0,0,0,0,0,0],
                          [0,0,0,0,0,0],
                          [0,0,0,0,0,0],
                          [0,0,0,0,0,0],
                          [0,0,0,0,0,0]];
        return (
            <div>
                <h1>
                    This is the solving page!
                </h1>
                <Puzzle topRow={top} sideRow={side} solveState={progress}></Puzzle>
            </div>
        );
    }
}

export default Solving;