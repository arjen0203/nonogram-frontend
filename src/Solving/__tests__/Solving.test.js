import React from "react";
import ReactDOM from 'react-dom';
import Solving from "../Solving";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"
import { act as actRenderer, create } from "react-test-renderer";
import { act } from 'react-dom/test-utils';
import GetNonogram from "../GetNonogram";


afterEach(cleanup);

it('Renders without crashing', function () {
	fetch.dontMockOnce();

	const div = document.createElement('div');
	ReactDOM.render(<GetNonogram></GetNonogram>, div);
});

it('Render correctly', async function () {
	const data = {
		"id": 210,
		"name": "boi",
		"topValues": [
			[1],
			[1, 1, 2],
			[1, 2, 2],
			[1, 1],
			[1, 1, 1],
			[1, 2, 1],
			[1, 1],
			[1, 2, 2],
			[1, 1, 2],
			[1]
		],
		"sideValues": [
			[10],
			[0],
			[2, 2],
			[1, 1],
			[1],
			[2],
			[1, 1],
			[2, 2],
			[6],
			[0]
		],
		"user": {
			"id": 17,
			"username": "DitIsDeTest"
		}
	};

	fetch.mockResponseOnce(JSON.stringify(data));

	let comp;
	await act(async () => {
		comp = render(
			<GetNonogram></GetNonogram>
		).getByTestId("getNonogram");
	});

	expect(comp).toHaveTextContent("6");
});

it('Render correctly', async function () {
	const data = {
		"id": 249,
		"name": "Panda",
		"topValues": [
			[4],
			[3],
			[2, 1],
			[1, 2, 1],
			[1, 1, 1],
			[1, 1, 1],
			[1, 2, 1],
			[2, 1],
			[3],
			[4]
		],
		"sideValues": [
			[10],
			[3, 3],
			[2, 2],
			[1, 1],
			[1, 1],
			[2, 2],
			[0],
			[2],
			[0],
			[4]
		],
		"user": {
			"id": 16,
			"username": "test123"
		}
	};

	fetch.mockResponseOnce(JSON.stringify(data));

	let comp;
	await act(async () => {
		comp = render(
			<GetNonogram></GetNonogram>
		).getByTestId("getNonogram");
	});

	expect(comp).toHaveTextContent("10");
});

it('Render correctly throw error', async function () {
	fetch.mockReject(new Error());

	let comp;
	await act(async () => {
		comp = render(
			<GetNonogram></GetNonogram>
		).getByTestId("getNonogram");
	});

	expect(comp).toHaveTextContent("Could not communicate with server");
});

it('Render nonogram not found', async function () {
	fetch.mockResponseOnce("Nonogram not found", {status: 404});

	let comp;
	await act(async () => {
		comp = render(
			<GetNonogram></GetNonogram>
		).getByTestId("getNonogram");
	});

	expect(comp).toHaveTextContent("Nonogram not found");
});


it('matches snapshot', async function () {
	const data = {
		"id": 210,
		"name": "boi",
		"topValues": [
			[1],
			[1, 1, 2],
			[1, 2, 2],
			[1, 1],
			[1, 1, 1],
			[1, 2, 1],
			[1, 1],
			[1, 2, 2],
			[1, 1, 2],
			[1]
		],
		"sideValues": [
			[10],
			[0],
			[2, 2],
			[1, 1],
			[1],
			[2],
			[1, 1],
			[2, 2],
			[6],
			[0]
		],
		"user": {
			"id": 17,
			"username": "DitIsDeTest"
		}
	};

	fetch.mockResponseOnce(JSON.stringify(data));

	let tree;
	await actRenderer(async () => {tree = create(<GetNonogram></GetNonogram>)});
	expect(tree.toJSON()).toMatchSnapshot();
});