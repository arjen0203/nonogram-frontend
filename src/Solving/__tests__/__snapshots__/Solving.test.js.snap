// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="getNonogram"
>
  <div
    className="puzzle"
  >
    <div>
      <div
        className="row"
      >
         
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            1
          </div>
          <div
            className="to-be-solved-top"
          >
            2
          </div>
        </div>
        <div
          className="top-number-column"
        >
          <div
            className="to-be-solved-top"
          >
            1
          </div>
        </div>
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          10
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="solved-side"
        >
          0
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          2
        </div>
        <div
          className="to-be-solved-side"
        >
          2
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          1
        </div>
        <div
          className="to-be-solved-side"
        >
          1
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          1
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          2
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          1
        </div>
        <div
          className="to-be-solved-side"
        >
          1
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          2
        </div>
        <div
          className="to-be-solved-side"
        >
          2
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="to-be-solved-side"
        >
          6
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
      <div
        className="row"
      >
         
        <div
          className="solved-side"
        >
          0
        </div>
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="square"
          data-testid="square"
          onContextMenu={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
         
      </div>
    </div>
  </div>
</div>
`;
